examples/flutter_gallery/lib/gallery/app.dart

```diff
@@ -153,7 +153,7 @@ class GalleryAppState extends State<GalleryApp> {
       // using named routes, consider the example in the Navigator class documentation:
       // https://docs.flutter.io/flutter/widgets/Navigator-class.html
       _kRoutes[item.routeName] = (BuildContext context) {
-        return item.buildRoute(context);
+        return new Semantics(routeName: item.routeName, child: item.buildRoute(context));
```

Comment:
Will developers always have to customize their routes with Semantics widget to get proper announcements and audio-feedback for route changes? Can the framework detect (some/alll?) route changes automatically and produce the appropriate feedback to ensure that most route changes are accessibility-friendly out-of-the-box?

Reply:
To account for the differences between material/cupertino - this should probably be done at the widget level.  For example, in MaterialRoute I can add a semantics which defaults to the route name, or "home" for the initial route.  This can be overriden by the user by providing a more specific semantics deeper in the tree, but the announcement couldn't ever be removed unless we added a special sentinel value or something.

Another approach is to add a field to `RouteSettings`.  That might allow us to automatically handle the case of just using the route name, but still allow users to prevent announcements from being made.  This would also handle all of the different Route implementations at once.

For widgets like the drawer, It is probably best just to add another named argument with a reasonable default.

Reply:
we could also handle the lack of announcement names in iOS by pushing a sentinel string by default for iOS platforms, and then excluding that for cupertino widgets which announce themselves.